src/Aspire.Hosting/Dashboard/DashboardOptions.cs

```diff
@@ -11,23 +12,42 @@ internal class DashboardOptions
 {
     public string? DashboardPath { get; set; }
     public string? DashboardUrl { get; set; }
+    public FrontendAuthMode DashboardAuthMode { get; set; }
     public string? DashboardToken { get; set; }
+    public OpenIdConnectOptions? OpenIdConnect { get; set; } = new();
+    public OpenIdConnectSettings? OpenIdConnectSettings { get; set; } = new();
     public string? OtlpGrpcEndpointUrl { get; set; }
     public string? OtlpHttpEndpointUrl { get; set; }
     public string? OtlpApiKey { get; set; }
     public string AspNetCoreEnvironment { get; set; } = "Production";
 }
 
+internal class OpenIdConnectSettings
+{
+    public string? Authority { get; set; }
+    public string? MetadataAddress { get; set; }
+    public string? ClientId { get; set; }
+    public string? ClientSecret { get; set; }
+    public ICollection<string> Scope { get; } = [];
+}
+
 internal class ConfigureDefaultDashboardOptions(IConfiguration configuration, IOptions<DcpOptions> dcpOptions) : IConfigureOptions<DashboardOptions>
 {
     public void Configure(DashboardOptions options)
     {
         options.DashboardPath = dcpOptions.Value.DashboardPath;
-        options.DashboardUrl = configuration["ASPNETCORE_URLS"];
-        options.DashboardToken = configuration["AppHost:BrowserToken"];
+        options.DashboardUrl = configuration[KnownConfigNames.AspNetCoreUrls];
 
-        options.OtlpGrpcEndpointUrl = configuration["DOTNET_DASHBOARD_OTLP_ENDPOINT_URL"];
-        options.OtlpHttpEndpointUrl = configuration["DOTNET_DASHBOARD_OTLP_HTTP_ENDPOINT_URL"];
+        if (Enum.TryParse<FrontendAuthMode>(configuration[DashboardConfigNames.DashboardFrontendAuthModeName.ConfigKey], out var dashboardAuthMode))
+        {
+            options.DashboardAuthMode = dashboardAuthMode;
+        }
+
+        options.DashboardToken = configuration["AppHost:BrowserToken"];
+        configuration.Bind("Dashboard:Frontend:OpenIdConnect", options.OpenIdConnect);
+        configuration.Bind("Authentication:Schemes:OpenIdConnect", options.OpenIdConnectSettings);
```

Comment:
I've been saying this for a while docs of hosting configuration is important for a while but now I must insist: It **_needs_** to be documented somewhere. As a reviewer who isn't an expert with hosting code I have no idea what configuration options the host accepts. Is this config accepted somewhere but no one knows about it? Is it following the naming standards of other docs? The current situation is how a big mess is created. Some order needs to be brought to this area.

And, not to meantion, docs need to be added so users know what options they can pass via hosting configuration.

Who is going to own adding it? I'll do it if no one else will...

Reply:
@JamesNK I'm not sure if this is exactly what you're looking for, but I did find this [dashboard configuration documentation](https://learn.microsoft.com/en-us/dotnet/aspire/fundamentals/dashboard/configuration?tabs=bash#common-configuration) that originally lead to this PR as it left me under the impression this was already a supported scenario.

Reply:
Dashboard configuration is different from app host configuration. People thinking they're the same is a common point of confusion.

I've started a gist to list possible configuration: https://gist.github.com/JamesNK/a22d95de1580f67abf348b1779c7006d
It will be the source of a future app host configuration doc.